clientconn_test.go

```diff
@@ -180,91 +180,63 @@ func TestCloseConnectionWhenServerPrefaceNotReceived(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error while listening. Err: %v", err)
 	}
-	defer server.Close()
+	var (
+		conn1, conn2 net.Conn
+		over         uint32
+	)
+	defer func() {
+		server.Close()
+		if conn1 != nil {
```

Comment:
This deserves a comment.  Why does what we were doing before not work (i.e. `defer conn1.Close()` in the goroutine)?

Reply:
This is to make cleanup easier. Otherwise we need to make sure the server goroutine closes the second connection only after the test is successful which requires another signaling mechanism.

Reply:
Comment in the code please?  If I saw this, I'd try to move it into the goroutine as a `defer conn1.Close()` like it used to be.

Based on your explanation, though...can we still do this for conn1?